### PR TITLE
Support Matrix notifications

### PR DESCRIPTION
**Pull request by binaryplease**
Closes #5486

### CODE REVIEW

File: health/notifications/alarm-notify.sh.in [hunk 1]
```diff
@@ -2208,6 +2275,7 @@ for state in "${SENT_EMAIL}" \
 	"${SENT_CUSTOM}" \
 	"${SENT_IRC}" \
 	"${SENT_AWSSNS}" \
+    "${SENT_MATRIX}" \
```
Comment (Ferroin) [inline health/notifications/alarm-notify.sh.in]: Indenting is off here, please make sure it matches up with the surrounding code.

Comment (binaryplease) [reply health/notifications/alarm-notify.sh.in]: Pushed a commit to resolve this. My editor was using the wrong whitespace characters

**Comment by cakrit** [comment]
@binaryplease I installed Riot to try it out, can you give me quickly a link on how I can find the home server and get a token? I couldn't find it quickly in their docs.

**Comment by binaryplease** [comment]
> @binaryplease I installed Riot to try it out, can you give me quickly a link on how I can find the home server and get a token? I couldn't find it quickly in their docs.

If you have set up your own homeserver use this:
```
curl -XPOST -d '{"type":"m.login.password", "user":"your_username", "password":"your_password"}' "https://localhost:8448/_matrix/client/r0/login"
```
on the server itself.
If you don't want to do that, you can just register a account at matrix.org and use this to get your token, the homeserver will be matrix.org in that case.

```
curl -XPOST -d '{"type":"m.login.password", "user":"your_username", "password":"your_password"}' "https://matrix.org/_matrix/client/r0/login"
```

File: health/notifications/alarm-notify.sh.in [hunk 2]
```diff
@@ -1664,6 +1675,55 @@ send_awssns() {
...
+        {
+            "msgtype": "m.notice",
+            "format": "org.matrix.custom.html",
+            "formatted_body": "<font color=\"$color\">${host} ${status_message}, \`${chart}\` (${family}), <b>${alarm}</b></font>",
+            "body": "${host} ${status_message}, \`${chart}\` (${family}), *${alarm}*"
+        }
+EOF
+		)"
+
+		httpcode=$(docurl -X POST --data-urlencode "payload=${payload}" "${webhook}")
```
Comment (cakrit) [inline health/notifications/alarm-notify.sh.in]: I finished testing.

I kept getting errors when I ran the curl command manually:

```
{
    "errcode": "M_NOT_JSON",
    "error": "Content not JSON."
}
```
I tried various things and the only way I could get it to work was to change the `--data-urlencode "payload=...` to `--data "..."` (no url encoding and no `payload=`).
The curl examples at https://matrix.org/docs/guides/client-server.html are a bit strange, but it looks like this is necessary.

File: health/notifications/alarm-notify.sh.in [hunk 3]
```diff
@@ -1664,6 +1675,55 @@ send_awssns() {
...
+
+	for room in ${rooms}; do
+        webhook="$homeserver/_matrix/client/r0/rooms/$(urlencode $room)/send/m.room.message?access_token=$accesstoken"
+		payload="$(
+        cat <<EOF
+        {
+            "msgtype": "m.notice",
+            "format": "org.matrix.custom.html",
+            "formatted_body": "<font color=\"$color\">${host} ${status_message}, \`${chart}\` (${family}), <b>${alarm}</b></font>",
+            "body": "${host} ${status_message}, \`${chart}\` (${family}), *${alarm}*"
```
Comment (cakrit) [inline health/notifications/alarm-notify.sh.in]: Can you add a link to ${goto_url} somewhere here? In a notification I received I saw (surprisingly) that family was converted to a link that leads nowhere. I didn't go any deeper into the documentation of the API, but it shouldn't be like that. Perhaps make the host or the alarm clickable and pass goto_url. You don't need to urlencode it, it's already ready for use. You can check other examples.

**Comment by cakrit** [comment]
Thanks @binaryplease. I think it's a good idea to copy/paste these commands you provided to README.md, as it wasn't that obvious to me.
I completed the review, please see if you can proceed with the requested fixes there. It should be quick and painless :)

**Comment by cakrit** [comment]
Kind reminder, will you be able to do the fixes @binaryplease ?

**Comment by cakrit** [comment]
Closing due to inactivity. Please comment if you wish to proceed and we'll reopen it.

**Comment by binaryplease** [comment]
@cakrit  Sorry for the inactivity, I just pushed some changes. Have not looked at this for a while, but I think it addresses the requests you made. Let me know if something else is missing!

**Comment by cakrit** [comment]
@binaryplease I reopened it, but it's been so long, that it requires another acceptance of the CLA and there are conflicts now. I will close it again, because it will be much simpler and cleaner to do a brand new PR now. My apologies for not catching the update earlier!

**Comment by ilyam8** [comment]
>  I will close it again, because it will be much simpler and cleaner to do a brand new PR now. My apologies for not catching the update earlier!

@pinpox any chance you can open a new PR?

**Comment by okias** [comment]
New PR is #9196 , hopefully last one :)

**Pull request closed without merging**